app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,61 @@
+class RecipesController < ApplicationController
+  before_action :authenticate_user!, except: %i[index show toggle]
+  before_action :set_recipe, except: %i[index new create]
+  def index
+    @recipe = Recipe.all
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def create
+    @recipe = Recipe.new(recipe_params)
+    @recipe.user = current_user
+    if @recipe.save
+      redirect_to recipes_path
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def show
+    @recipe_foods = @recipe.recipe_foods
+  rescue ActiveRecord::RecordNotFound
+    redirect_to recipes_path
+  end
+
+  def destroy
+    @recipe.destroy
+    redirect_to recipes_path
+  end
+
+  def update
+    @recipe.toggle(:public).save
```

Comment:
- [ ] Great job working on this capstone project. You are almost there. Please kindly check your toggle for public recipes. It seems not working. At the moment all added recipes are public. I could not get any recipes off the public display